locksmith/src/operations/stripeOperations.ts

```diff
@@ -59,7 +59,35 @@ export const saveStripeCustomerIdForAddress = async (
   }
 }
 
+/**
+ * Method which delets the stripe customer id for an address
+ */
+export const deletePaymentDetailsForAddress = async (
+  publicKey: ethereumAddress
+) => {
+  const normalizedEthereumAddress = Normalizer.ethereumAddress(publicKey)
+
+  // First, let's delete the StripeCustomer
+  const deletedStripeCustomer = await StripeCustomer.destroy({
+    where: { publicKey: { [Op.eq]: normalizedEthereumAddress } },
+  })
+
+  // Then, update UserReference
+  // TODO: deprecate when all stripe_customer_id UserReferences have been moved to use StripeCustomer
+  const [updatedUserReference] = await UserReference.update(
+    {
+      stripe_customer_id: '',
```

Comment:
Not sure if this is being set to empty string due to a table constraint not allowing  NULL or what but this generally creates down stream pain where people need to have knowledge about this rather than a value existing or being NULL.

If possible we should set this to NULL

Reply:
good point. let me check if NULL is possible.